internal/provider/resource_gitlab_service_emails_on_push.go

```diff
@@ -0,0 +1,167 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_service_emails_on_push", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_service_emails_on_push`" + ` resource allows to manage the lifecycle of a project integration with Emails on Push Service.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/integrations.html#emails-on-push)`,
+
+		CreateContext: resourceGitlabServiceEmailsOnPushCreate,
+		ReadContext:   resourceGitlabServiceEmailsOnPushRead,
+		UpdateContext: resourceGitlabServiceEmailsOnPushCreate,
+		DeleteContext: resourceGitlabServiceEmailsOnPushDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Description:  "ID of the project you want to activate integration on.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"recipients": {
+				Description:  "Emails separated by whitespace.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"disable_diffs": {
+				Description: "Disable code diffs.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"send_from_committer_email": {
+				Description: "Send from committer.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"push_events": {
+				Description: "Enable notifications for push events.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"tag_push_events": {
+				Description: "Enable notifications for tag push events.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"branches_to_be_notified": {
+				Description: `Branches to send notifications for. Valid options are "all", "default", "protected", and "default_and_protected". Notifications are always fired for tag pushes.`,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "all",
```

Comment:
🚧 **Suggestion:** let's add a validator here, similar to what we have here:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/96e5f299d937aa1d74281aa21cbb92938f286cbe/internal/provider/resource_gitlab_group_access_token.go#L76

Also make sure to render the options accordingly in the `Descriptions`. Also see the above example.

Reply:
Cool, I was suspecting there is a nicer way to describe this :) Thanks for pointing out